src/core/send_buffer.c

```diff
@@ -187,34 +187,63 @@ QuicSendBufferFill(
     }
 }
 
+_IRQL_requires_max_(DISPATCH_LEVEL)
+uint32_t
+QuicGetNextIdealBytes(
+    _In_ uint32_t BaseValue
+    )
+{
+    uint32_t Threshold = QUIC_DEFAULT_IDEAL_SEND_BUFFER_SIZE;
+
+    //
+    // We calculate the theshold as an exponential grown based on the default
+    // up to the max, rounding up to the next threshold when equal.
+    //
+    while (Threshold <= BaseValue) {
```

Comment:
I think "==" makes more sense, because we want to avoid underestimation, as opposed to wanting overestimation.

Reply:
Sorry, I meant "<" :) And actually, it looks like your design requires "<=", since BaseValue (when it's BytesInFlightMax) is limited by the threshold value.